### PR TITLE
Bug 1585252 - Fix string encoding errors for sync with Python 3.

### PR DESCRIPTION
**Pull request by adngdb**
@jotes mind giving this a quick look too, please?

### CODE REVIEW

File: pontoon/sync/vcs/repositories.py [hunk 1]
```diff
@@ -327,6 +330,8 @@ def normalize_revision(rev):
             ['svn', 'diff', '-r', '{}:{}'.format(from_revision, 'HEAD'), '--summarize'],
             cwd=path
         )
+        if type(output) is bytes:
```
Comment (jotes) [inline pontoon/sync/vcs/repositories.py]: Do you want to drop this on Python 3?

Comment (adngdb) [reply pontoon/sync/vcs/repositories.py]: I'm not sure, I've seen cases where this was an instance of `str` and I wasn't able to figure out when or why.

File: pontoon/sync/vcs/repositories.py [hunk 2]
```diff
@@ -173,6 +173,9 @@ def commit(self, path=None, message=None, user=None, branch=None):
         if code != 0:
             raise CommitToRepositoryException(text_type(error))
 
+        if type(error) is bytes:
+            error = error.decode('utf-8')
```
Comment (mathjazz) [inline pontoon/sync/vcs/repositories.py]: Why do you only add the check for the `commit()` method of the `CommitToGit` class, and not also other `CommitToX` classes?

Comment (adngdb) [reply pontoon/sync/vcs/repositories.py]: Ah, that's because this is the only one I tested for, and I didn't think to look at the other classes. Thanks!

**Comment by mathjazz** [comment]
I guess we should do the same for SVN, then?

What about `PullFromX.pull()`?

**Comment by adngdb** [comment]
@mathjazz this is a lot better tested now. I was able to sync all files from `pontoon-test` and write to all file formats as well — see https://github.com/adngdb/pontoon-test. Can you review again?

**Pull request merged**